gateway/src/test/java/io/zeebe/mocks/RequestFutureErrorMock.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.zeebe.mocks;
+
+import io.zeebe.broker.client.api.ZeebeFuture;
+import io.zeebe.broker.client.api.commands.Topology;
+import io.zeebe.broker.client.api.commands.TopologyRequestStep1;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class RequestFutureErrorMock<E> implements ZeebeFuture<E>, TopologyRequestStep1 {
```

Comment:
You have a generic but which then implements `TopologyRequestStep1` which is a specific type. If you are just interested in the `send` method then implement `FinalCommandStep<E>` instead. Then it is reusable for other tests.

Reply:
Yup, I'll change this. I was not sure how I will end up doing the thing I wanted to do so it ended up like this.

Reply:
It doesn't work. This would also mean that we can't take the shortcuts which I'm doing and then we would have to mock metric ton of stuff. I see you are complaining a lot about generated stuff here so I will just rewrite it.